src/main/java/net/jonathangiles/tools/teenyhttpd/json/TeenyJson.java

```diff
@@ -0,0 +1,377 @@
+package net.jonathangiles.tools.teenyhttpd.json;
+
+
+import net.jonathangiles.tools.teenyhttpd.util.ParameterType;
+import net.jonathangiles.tools.teenyhttpd.util.ReflectionUtils;
+
+import java.beans.Transient;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Parameter;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * 0 dependencies Teeny JSON serializer
+ * <p>
+ * TODO: add support for Maps/Dictionaries
+ * TODO: remove all static methods and make it a proper class to enable custom serialization
+ * @author alex
+ */
+public class TeenyJson {
+
+    public static String write(Object object) {
```

Comment:
write implies this is going to an output stream. I wonder if `toString` is better?

Reply:
What about `writeValue` or `writeAsString` their counter part would be `readValue(String)`? `toString` will not be the best option after removing all the static methods

Examples

```java
String json = new TeenyJson()
   .writeValue(object); 
```

```java
byte[] json = new TeenyJson()
   .serializer(Boolean.class, new CustomBooleanSerializer())
   .writeValue(StandardCharsets.ISO_8859_1, object);
```

```java
new TeenyJson()
    .writeValue(baos, object);
```

```java
Set<Pet> petSet = new TeenyJson()
    .readCollection(json, Set.class, Pet.class);
```


```java
Pet pet = new TeenyJson()
   .deserializer(Boolean.class, new BooleanIntCoercionDeserializer())
    .readValue(json, Pet.class);
```

```java
Pet pet = new TeenyJson()
    .readValue(baos, Pet.class);
```


pretty similar to Jackson

Reply:
Let's discuss in a new pr for Json support